refactor/improve-retrofit

Mostafa-khaled21 opened the pull request:
key changes include standardizing naming conventions for Retrofit builders, replacing hardcoded strings with constants for API endpoints, and reorganizing the Koin network module for improved clarity.

Mostafa-khaled21 committed:
Refactor: Network module configuration

- Renamed `networkRetrofitModule` to `networkModule`.
- Moved Retrofit client creation logic to `createRetrofitClient.kt`.
- Updated module to use constants for base URL and access token.
- Standardized qualifiers for Retrofit builders.
- Simplified Retrofit service provisioning.

Mostafa-khaled21 committed:
Refactor: Define constants for Retrofit API endpoints and parameters

Mostafa-khaled21 committed:
Refactor: Rename Retrofit builder instances for consistency

The variable names for instances of `RetrofitRequestBuilder` and `UserApiServiceRetrofit` have been changed from uppercase to lowercase for consistency across the following classes:

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/develop' into refactor/improve-retrofit

# Conflicts:
#	datasource/remote/media/src/main/java/com/giraffe/media/person/retrofit/PersonApiServiceRetrofit.kt
#	datasource/remote/media/src/main/java/com/giraffe/media/person/retrofit/PersonRemoteDataSourceImpRetrofit.kt

Mostafa-khaled21 committed:
Refactor: Update person search API to support pagination

mohamedebrahem13 commented on app/src/main/java/com/giraffe/cineverseapp/di/networkModule.kt:
```diff
@@ -61,46 +63,40 @@ val networkRetrofitModule = module {
+        SeriesRemoteRetrofitDataSourceImp(get(named(QUALIFIER_SERIES_BUILDER)))
     }
 
     single(named("user_retrofit")) {
```
You are duplicating Retrofit Builders unnecessarily, by doing this you will have two singletons in the memory for retrofit throw the app runtime, also the base URL is the same you don't need this

Mostafa-khaled21 committed:
Refactor: Simplify UserApiServiceRetrofit instantiation

Removed the named qualifier and separate Retrofit instance for `UserApiServiceRetrofit`, now using the default Retrofit instance provided by `provideRetrofitService`.

mohamedebrahem13 reviewed (dismissed):
great work 💯

skaik-mo commented on app/src/main/java/com/giraffe/cineverseapp/data/network/createRetrofitClient.kt:
```diff
@@ -0,0 +1,22 @@
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+
+fun createRetrofitClient(accessToken: String): OkHttpClient {
```
The file name should follow camel case convention

Mostafa-khaled21 committed:
changed the file name to camel case

poula22 reviewed (approved):
Great job👏💯
